src/anomalib/models/image/glass/lightning_model.py

```diff
@@ -0,0 +1,326 @@
+            Defaults to `0.015`.
+        radius (float, optional): Radius parameter used for truncated projection in the anomaly synthesis strategy.
+            Determines the range for valid synthetic anomalies in the hypersphere or manifold.
+            Defaults to `0.75`.
+        p (float, optional): Probability used in random selection logic, such as anomaly mask generation or augmentation
+          choice.
+            Defaults to `0.5`.
+        lr (float, optional): Learning rate for training the feature adaptor and discriminator networks.
+            Defaults to `0.0001`.
+        step (int, optional): Number of gradient ascent steps for anomaly synthesis.
+            Defaults to `20`.
+        svd (int, optional): Flag to enable SVD-based feature projection.
+            Defaults to `0`.
+    """
+
+    def __init__(
+        self,
+        input_shape: tuple[int, int],
+        anomaly_source_path: str,
+        backbone: str = "resnet18",
+        pretrain_embed_dim: int = 1024,
+        target_embed_dim: int = 1024,
+        patchsize: int = 3,
+        patchstride: int = 1,
+        pre_trained: bool = True,
+        layers: list[str] | None = None,
+        pre_proj: int = 1,
+        dsc_layers: int = 2,
+        dsc_hidden: int = 1024,
+        dsc_margin: float = 0.5,
+        pre_processor: PreProcessor | bool = True,
+        post_processor: PostProcessor | bool = True,
+        evaluator: Evaluator | bool = True,
+        visualizer: Visualizer | bool = True,
+        mining: int = 1,
+        noise: float = 0.015,
+        radius: float = 0.75,
+        p: float = 0.5,
+        lr: float = 0.0001,
+        step: int = 20,
+        svd: int = 0,
+    ) -> None:
+        super().__init__(
+            pre_processor=pre_processor,
+            post_processor=post_processor,
+            evaluator=evaluator,
+            visualizer=visualizer,
+        )
+
+        if layers is None:
+            layers = ["layer1", "layer2", "layer3"]
+
+        self.augmentor = PerlinAnomalyGenerator(anomaly_source_path)
+
+        self.model = GlassModel(
+            input_shape=input_shape,
+            anomaly_source_path=anomaly_source_path,
+            pretrain_embed_dim=pretrain_embed_dim,
+            target_embed_dim=target_embed_dim,
+            backbone=backbone,
+            pre_trained=pre_trained,
+            patchsize=patchsize,
+            patchstride=patchstride,
+            layers=layers,
+            pre_proj=pre_proj,
+            dsc_layers=dsc_layers,
+            dsc_hidden=dsc_hidden,
+            dsc_margin=dsc_margin,
+            step=step,
+            svd=svd,
+            mining=mining,
+            noise=noise,
+            radius=radius,
+            p=p,
+        )
+
+        self.c = torch.tensor([1])
+        self.lr = lr
+
+        self.dev = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
@hgaiser, there is a bug here to be fixed. For some reason, gpu is not detected for this model, which is why @code-dev05 had to add this line. However, this is temporary and not generic enough to run on any other GPUs (Intel, AMD or mac)